### DISCUSSION TITLE
Global option to skip test files which fail on imports

### DISCUSSION THREAD

**Discussion by ElArkk** [pytest-dev/pytest#10570] [Q&A]
I'm wondering whether there is a more high-level way of skipping whole test files due to failing imports than `pytest.importorskip`.

The reason that I'd need such a "global" option is twofold:
- I have tests that can fail due to missing imports in imported dependencies, which makes it tedious to capture those in the test files themselves using `importorskip`.
- I have flake8 and isort pre-commit hooks enforcing all imports being at the top of the file, so I'd need to exempt all files using `importorskip` from those hooks. As I need to put `importorskip` before importing the dependencies that might fail due to those missing imports.

**Comment by portyu9** [comment]
Hi @ElArkk 

There is no global pytest option that safely turns **every collection-time `ImportError`** into a skipped test module.

That distinction is intentional: an import failure may mean an optional dependency is absent, but it may also mean your package is genuinely broken. Automatically converting both cases into skips could hide real failures.

For an optional dependency, the supported approach is still:

```python
import pytest

optional_package = pytest.importorskip("optional_package")
```

at module level, before importing code which requires it.

Current pytest also has:

```text
--continue-on-collection-errors
```

but that only allows the rest of the test suite to continue. The import failure remains a **collection error** and the pytest run still fails; it does not turn the module into a skip.

If an entire directory is only applicable when an optional dependency exists, another option is to exclude that directory during collection from `conftest.py`, but then those tests are ignored rather than reported as skipped.

So for optional dependencies where you want a proper `SKIPPED` result, `pytest.importorskip()` remains the appropriate solution.

Please mark this answer as accepted if it helped, thank you.
